src/js/login/containers/Main.jsx

```diff
@@ -32,14 +55,18 @@ export class Main extends React.Component {
     if (event.data === sessionStorage.userToken) { this.props.getUserData(); }
   }
 
+  getRedirectUrl = () => (new URLSearchParams(window.location.search)).get('next');
+
   bindNavbarLinks = () => {
     [...document.querySelectorAll('.login-required')].forEach(el => {
       el.addEventListener('click', e => {
-        e.preventDefault();
-        const nextQuery = { next: el.getAttribute('href') };
-        const nextPath = appendQuery('/', nextQuery);
-        history.pushState({}, el.textContent, nextPath);
-        this.props.toggleLoginModal(true);
+        if (!this.props.login.currentlyLoggedIn) {
```

Comment:
If we know the user is logged in, no need to send to "?next=..." for `componentDidUpdate` to then redirect accordingly. Can just treat it as a normal link in that case.